plugins/javaembed/javaembed.cpp

```diff
@@ -2856,56 +2350,312 @@ class JavaEmbedImportContext : public CInterfaceOf<IEmbedFunctionContext>
                 if (stricmp(optName, "classpath")==0)
                     classpath.set(val);
                 else
-                    throw MakeStringException(0, "javaembed: Unknown option %s", optName.str());
+                    throw MakeStringException(0, "javaembed: Unknown option %s", optName.str());
+            }
+        }
+    }
+    ~JavaEmbedImportContext()
+    {
+        if (javaClass)
+            JNIenv->DeleteGlobalRef(javaClass);
```

Comment:
does instance need to be cleaned up?

Reply:
If I knew what instance was I might agree...

Reply:
Looks like it's used by the code that embeds Java code into ESP services, and that it is actually owned by that code. So no.